rust/datafusion/src/execution/context.rs

```diff
@@ -426,6 +430,51 @@ impl ExecutionContext {
             }
         }
     }
+
+    /// Execute a query and write the results to a partitioned CSV file
+    pub fn write_csv(&self, plan: &dyn ExecutionPlan, path: &str) -> Result<()> {
+        // create directory to contain the CSV files (one per partition)
+        let path = path.to_string();
+        println!("Creating {}", path);
+        fs::create_dir(&path)?;
+
+        let threads: Vec<JoinHandle<Result<()>>> = plan
+            .partitions()?
+            .iter()
+            .enumerate()
+            .map(|(i, p)| {
+                let p = p.clone();
+                let path = path.clone();
+                thread::spawn(move || {
+                    let filename = format!("part-{}.csv", i);
+                    let path = Path::new(&path).join(&filename);
+                    println!("Writing {}", path.to_str().unwrap());
+                    let file = fs::File::create(path)?;
+                    let mut writer = csv::Writer::new(file);
```

Comment:
Are you planning on exposing write options in future (delimiter, headers)?

Reply:
Yes, I think we should, in a future PR.